[Onboarding] Questionnaire

mennenia opened the pull request:
As per #1389 most of it is there, apart from the ability to go back. 
Also, couldn't get grey borders on the button just yet, will refine. 


![pr](https://cloud.githubusercontent.com/assets/373860/14401383/64767226-fdde-11e5-85e7-273af32cb9a2.GIF)

alloy commented on Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.h:
```diff
@@ -5,10 +5,12 @@
 - (void)splashDone:(ARSignUpSplashViewController *)sender;
 - (void)splashDoneWithLogin:(ARSignUpSplashViewController *)sender;
-- (void)collectorLevelDone:(ARCollectorLevel)level;
+//- (void)collectorLevelDone:(ARCollectorLevel)level;
```
This is just temporarily disabled, I assume?

alloy commented on Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.h:
```diff
@@ -23,6 +25,17 @@ typedef NS_ENUM(NSInteger, ARInitialOnboardingState) {
+typedef NS_ENUM(NSInteger, AROnboardingStage) {
+    AROnboardingStageSlideshow,
+    AROnboardingStageStart,
+    AROnboardingStageChooseMethod,
```
This could probably use some clarification love too; choose what method?

mennenia replied:
No clue, it's been copied from what was there before. I kept some (like this one) to not break the existing code, but it will all be replaced/clarified as I work on the different areas.

alloy commented on Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.m:
```diff
@@ -246,89 +239,85 @@ - (void)didSignUpAndLogin
 }
 
-- (void)collectorLevelDone:(ARCollectorLevel)level
+- (void)presentPersonalize
```
Feel free to make these names more self-explanatory, which I much prefer. E.g. `presentPersonalizationQuestionnaires`.

mennenia replied:
yup

alloy commented on Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/Views/AROnboardingHeaderView.m:
```diff
@@ -0,0 +1,62 @@
+
+@property (nonatomic) UIView *progressBar;
+@property (nonatomic) UIView *progressBackgroundBar;
+@property (nonatomic) UILabel *titleLabel;
```
The existing code-base has a strong preference for listing all the property attributes, e.g. `strong, readwrite`.

mennenia replied:
will fix

alloy commented on Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/Views/AROnboardingNavigationItemsView.m:
```diff
@@ -0,0 +1,82 @@
+        [_warningLabel constrainWidthToView:self predicate:@"0"];
+        [_warningLabel constrainHeight:@"30"];
+        [_warningLabel alignLeadingEdgeWithView:self predicate:@"0"];
+        [_warningLabel constrainBottomSpaceToView:_back predicate:@"0"];
```
You may want to move setting constraints to `-[UIView updateConstraints]` for readability purposes, but seeing as you don’t ever change any of the constraints it’s not a blocker  imo.

mennenia replied:
sounds good will do

alloy commented:
This is look sweet, well done and thanks for the screen recording 👍 

From looking at the screen recording, it seems like something goes wrong with the splash screen background, which turns black on transition.

mennenia commented:
Yup noticed that, will also be fixed!

mennenia commented:
![pr2](https://cloud.githubusercontent.com/assets/373860/14447459/f902b994-002d-11e6-8a1f-69ba66c71e58.GIF)

Going back is in there + addressed comments.

I know code wise it's a bit rough, but that's because I purposefully choose to split this from the search part which is what actually completes the VC and most of the onboarding so I can clean up the code and add the proper tests.
